### PR TITLE
Implement MSC 3981

### PR DESCRIPTION
**Pull request by justjanne**
Type: Enhancement
Fixes: https://github.com/vector-im/element-web/issues/25021
Related: https://github.com/matrix-org/matrix-spec-proposals/pull/3981

---
Here's what your changelog entry will look like:

## ✨ Features
 * Implement MSC 3981 ([\#3248](https://github.com/matrix-org/matrix-js-sdk/pull/3248)). Fixes vector-im/element-web#25021. Contributed by @justjanne.

### CODE REVIEW

File: src/@types/requests.ts [hunk 1]
```diff
@@ -186,6 +186,7 @@ export interface IRelationsRequestOpts {
     to?: string;
     limit?: number;
     dir?: Direction;
+    recurse?: boolean;
```
Comment (t3chguy) [inline src/@types/requests.ts:189]: A comment linking it to MSC3981 would be good

File: src/models/thread.ts [hunk 2]
```diff
@@ -458,25 +459,27 @@ export class Thread extends ReadReceipt<EmittedEvents, EventHandlerMap> {
...
-                            if (relations.events.length) {
-                                event.makeReplaced(relations.events[0]);
-                            }
-                        })
-                        .catch((e) => {
-                            logger.error("Failed to load edits for encrypted thread event", e);
-                        });
-                }),
-        );
+        if (!this.client.canSupport.get(Feature.RelationsRecursion)) {
```
Comment (t3chguy) [inline src/models/thread.ts:462]: This looks wrong, shouldn't it be `!== Unsupported` - right now it says if not stable, so doesn't allow unstable behaviour.

Comment (justjanne) [reply src/models/thread.ts:462]: You're right! Here I actually made the very error that I used as justification for why we shouldn't coerce values to booleans like this^^

Comment (t3chguy) [reply src/models/thread.ts:462]: https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/strict-boolean-expressions.md can prevent it ftr

**Review by t3chguy — APPROVED** [review]

**Pull request merged**